ui/src/sources/components/DashboardStep.tsx

```diff
@@ -52,19 +55,40 @@ interface Props {
 
 @ErrorHandling
 class DashboardStep extends Component<Props, State> {
+  private isComponentMounted: boolean
+
   public constructor(props: Props) {
     super(props)
     this.state = {
       selected: this.selected,
       protoboards: [],
       searchTerm: '',
       suggestedProtoboards: [],
+      fetchingProtoboards: RemoteDataState.NotStarted,
+      fetchingSuggested: RemoteDataState.NotStarted,
     }
   }
 
   public async componentDidMount() {
-    const protoboards = await getProtoboards()
-    this.setState({protoboards}, this.handleSuggest)
+    this.isComponentMounted = true
+
+    this.setState({fetchingProtoboards: RemoteDataState.Loading})
+    try {
+      const protoboards = await getProtoboards()
+
+      if (this.isComponentMounted) {
```

Comment:
is it possible for this.isComponentMounted to be false here? I couldn't see a way

Reply:
yes, if you clicked next in wizard before the request finished, it would call setstate on an unmounted component